test/multipart/test_decoder.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+
+import io
+import typing
+import unittest
+from unittest import mock
+
+import pytest
+
+import urllib3.response
+from urllib3.multipart.decoder import (
+    BodyPart,
+    ImproperBodyPartContentError,
+    MultipartDecoder,
+    NonMultipartContentTypeError,
+)
+from urllib3.multipart.encoder import MultipartEncoder, encode_with
+
+
+class TestBodyPart(unittest.TestCase):
```

Comment:
This and the test case in `test_encoder.py` are the only tests in the project that subclass `unittest.TestCase`.

I'm going to leave them as-is for now, but I wonder if they should be rewritten to use pytest fixtures or similar, to match the rest of the project.